File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-5.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
    int n = cost.length;
     int[] f = new int[n + 1];
    // f[i]=min(f[i−1]+cost[i−1],f[i−2]+cost[i−2])
    for (int i = 2 ; i <= n; i++) {
         f[i] = Math.min(f[i - 1] + cost[i - 1], f[i - 2] + cost[i - 2]);
    }
    return f[n];
 }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        min_value = 101
        
        for p in prices:
            min_value = min(min_value, p)
            profit = max(profit, p - min_value)
        return profit    
            
        